### DISCUSSION TITLE
Finding a good layout - General Practice

### DISCUSSION THREAD

**Discussion by ageekhere** [CorsixTH/CorsixTH#2404] [General Diagnostics]
I am exploring a good design layouts for each room, in this post I am focusing on the General Practice room.
**The goal**
Keeping the room as small as possible while keeping the doctor as happy as possible(avoid being unhappy and wanting a pay rise) while optimizing turnaround time for handyman plant watering and patient walk times to chair and door.

![Screenshot 2023-09-21 114014](https://github.com/CorsixTH/CorsixTH/assets/13173133/2867fea0-8ffc-4e83-8015-295406074fc7)

Here is my current idea.
size 5x6
1x desk (center of room)
1x filing cabinet (Placed right of desk for fast access)
1x chair (Placed Infront of door to minimize patient walk time)
1x radiator (behind desk where doctor is most of the time)
1x fire extinguisher (in corner, out of the way)
1x plant (next to door for quick access for handy man)
1x bin (next to door in path of patients so they can access it when walking to chair)
13x windows (however if made next to a wall this can change)

Realistic improvements that can be made to this design
1. Add more fire extinguishers (do more fire extinguishers improve happiness?)
2. Add more plants (longer handyman watering times)
3. Make room bigger

So what realistic improvements can be made to improve this layout while keeping in mind of the goal?

**Update 2**
After @MandrakeHorse findings on North and south facing desks I have updated the GP's Office, Yes the way the desks face make the doctor walk around his chair wasting time.

![Screenshot 2023-09-23 081316](https://github.com/CorsixTH/CorsixTH/assets/13173133/ce708e92-fb59-4b73-9153-b365631aff65)
The changes 
1. Desk now faces north west to avoid chair walk around
2. Room size change, now 4x5
3. Added extra radiator, one on each side of the room
![Screenshot 2023-09-23 081903](https://github.com/CorsixTH/CorsixTH/assets/13173133/9da6765b-db88-4269-9a71-4679bd210ea1)
4. Still 1 fire extinguisher, next to desk now
5. Plant is still in left of door
6. Bin is right of door
7. Chair is in front of door

Note on chair position, I think it is faster having the chair placed facing south east of the door for patient walk time as the patient does not need to make a turn to the chair. At the moment the patient opens the door, walks to the chair, turns around and sits down. If it is left or right of the door the patient needs to make a turn.

Summary on update 2.
1. The desk and filing cabinet positions are now the most optimal (no walk around chair) 
2. Room should have a better heat map with 2 radiators and being centered in the middle of the room works well.
3. Plant and bin position cannot be switched around.

Notes, there is still a little room left in the room in the east corner that something could be added (maybe an extra fire extinguisher or 2?).

**Update 3**
So the fire extinguisher has a 2 tile radius, with the the old design the doctor will walk out of the radius every time they go to the filing cabinet. To improve this i have moved the fire extinguisher to cover this. I have also added another radiator.
![Screenshot 2023-09-24 093919](https://github.com/CorsixTH/CorsixTH/assets/13173133/2ee0e1fa-b0fc-4c85-be43-89541db6de17)

**Comment by MandrakeHorse** [comment]
[I found this strategy guide on GameFAQs](https://gamefaqs.gamespot.com/pc/198993-theme-hospital/faqs/68892) which has a section on how to optimise the GP's office. Something that I found interesting is that it specifically tells you to place the desk so that it faces either **north-east** or **north-west**. For some reason (and I was able to confirm this behaviour in Corsixth), if the desk faces either south-east or south-west, the doctor will always get up and sit down using opposite sides of the desk, thus wasting time by walking round the chair.

In this first screenshot, you can see the doctors with north-facing desks getting up on the side of the desk which has the telephone and lamp. They walk to the adjacent filing cabinet, open and close the drawer, and then sit down at their desk from the same side.
![2023-09-22 21_40_42-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/142453312/712b73d6-3082-4c30-9ee4-b0a07b041e9a)
In this second screenshot, the doctors with south-facing desks have gotten up from the side of the desk with the computer. They will walk to the filing cabinet, check it, and then sit down from the side with the lamp. Even after I moved the filing cabinet to the opposite side, they would _always_ walk around the chair before sitting back down.
![2023-09-22 21_40_51-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/142453312/94b4d0d3-e52c-451b-8288-0b19299d0a3d)

**Reply by ageekhere** [reply]
I have updated the design for desk position.

**Comment by PaveMentman** [comment]
---
In regards to keeping the staff happy:
I personally simply try to give enough other amenities ( including trash-bins and green-plants )
alongside adjusting the policies / keeping the "fatigue"-percentage not above 50% thus the staff can have more fun-time at the break-rooms.
Oh and also making sure the whole facility is heated enough thus not damp.
So far this has been enough to keep my staff happy without asking for pay-raises,
especially since also keep an eye on the "census"-indicators
( too-cold, not enough green-plants, etc. complaints ).


Hence going far the smallest GP-office and other diagnosis plus final-treatment-rooms-strategy still works the best to me at least.


- Adding all-the-windows naturally is simply to allow seeing through walls with less of hassle.

- Fire extinguisher for the most part seem to simply cosmetic when placed outside rooms that have a malfunctionable-machines
( feel free to correct me if the info I have is incorrect ).
But considering it can simply be put into a corner or otherwise against a wall without occupying / obstructing the walk-path of the square,
the 25-game-money-units-cost is really nothing and it still adds a nice eye-candy;
plus it might add some realism since nowadays a lot of tral-life-office-rooms have a 1-kilogram-extinguisher which can cost less than 15€ per unit ( and this is the VAT-inclusive-price ).

- Since all rooms are meant to as small plus as optimized as possible accordingly to level / hospital-layout,
one green-plant per a room should be enough for the refreshment;
at least so far I've yet to encounter any patients vomiting in non-corridor-areas;
that is, if there is any coding allowing them to vomit "outside" corridors to begin with,
nevermind if my childhood-info actually holds true anymore when it comes to prevention of vomit-waves.


Now that I am done with the general-ramblings,
back on to the on to the actual GP-office layout-examples:
Here are hopefully my most optimised GP-offices next to the entrance of Frimpton-On-Sea / Level-4.


![2023-09-23-00-16-10-423-CorsixTH-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/6709463/e9dcb414-c7a7-4d78-87ba-fe6e8f0458d5)
![2023-09-23-00-16-13-775-CorsixTH-CorsixTH](https://github.com/CorsixTH/CorsixTH/assets/6709463/c659b4b0-c76c-4ae7-b8e8-b4b94ea8f61d)
---

**Reply by MandrakeHorse** [reply]
Your GP offices are nearly optimised, but the filing cabinet should be next to the desk's telephone/lamp so that your doctors don't have to constantly walk around their chairs and back,

**Reply by ageekhere** [reply]
If you look at my update 2 design, look at chair, plant and bin placing. 
I think having the chair placed left or right of the door makes the patient make a turn and then sit rather then just walking in and turning around to sit.

**Comment by PaveMentman** [comment]
> Your GP offices are nearly optimised, but the filing cabinet should be next to the desk's telephone/lamp so that your doctors don't have to constantly walk around their chairs and back,

Very true
It is worth noting that these rooms actually build few weeks ago (screenshots are fresh though).
And now I was reintroduced to these optimization-images and guides.
So for the time being I was totally okay sacrificing the few (or more) seconds for the filing cabinet shuffle in this specific scenario
( especially since I had finally re-found the walk-sit-paths after some trial-and-error ).

> If you look at my update 2 design, look at chair, plant and bin placing. I think having the chair placed left or right of the door makes the patient make a turn and then sit rather then just walking in and turning around to sit.

Ah yes, there was also that little quirk I had forgotten from the old guides;
thank you for the reminder.

I am currently waiting for the "pause-double-door-crash"-bug-fix,
so no new retry-images from me for the time being.
This is the bug I reference to:
https://github.com/CorsixTH/CorsixTH/issues/2389
